firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsController.java

```diff
@@ -375,8 +374,11 @@ public Task<Void> then(@Nullable Boolean send) throws Exception {
                       public Task<Void> call() throws Exception {
                         if (!send) {
                           Logger.getLogger().v("Deleting cached crash reports...");
-                          deleteFiles(listAppExceptionMarkerFiles());
-                          reportingCoordinator.removeAllReports();
+                          diskWriteWorker.submit(
```

Comment:
Do we rely on the fact files hae been deleted before `unsentReportsHandled` resolves?

If so, maybe this can include the `unsentReportsHandled.trySetResult(null);` in the task submitted to the disk worker, then return the result task? That way unsentReportsHandled will get resolved after the files have been deleted from disk.

Reply:
Assuming if we doing `unsentReportsHandled.trySetResult(null);` under diskWriter it will be resolved later right? I am a bit afraid the risk condition where while deleting the file, unsentReportsHandled still shows there is unsend report, when user check the if there is any unsend report we still show the status is true. For me this status is a status in memory, so regardless if the file is deleted or not, we don't care the status on disk anymore. By doing this we can improve efficiency of concurrency.